nat-lab/tests/test_vpn.py

```diff
@@ -436,45 +428,35 @@ async def connect(
             )
             alpha_ip = testing.unpack_optional(alpha.get_ip_address(ip_proto))
 
-            await exit_stack.enter_async_context(
-                connection.create_process([
-                    "nc",
-                    "-nv",
-                    "-6" if setup_params.ip_stack == IPStack.IPv6 else "-4",
-                    "-s",
-                    alpha_ip,
+            nc_client_1 = await exit_stack.enter_async_context(
+                NetCatClient(
+                    connection,
                     serv_ip,
-                    str(80),
-                ]).run(
-                    stdout_callback=output_notifier.handle_output,
-                    stderr_callback=output_notifier.handle_output,
-                )
+                    80,
+                    ipv6=ip_proto == IPProto.IPv6,
+                    source_ip=alpha_ip,
+                ).run()
             )
 
             # Second client, this time sending some data to check proper TCP sequence number generation
-            proc = await exit_stack.enter_async_context(
-                connection.create_process([
-                    "nc",
-                    "-nv",
-                    "-6" if setup_params.ip_stack == IPStack.IPv6 else "-4",
-                    "-s",
-                    alpha_ip,
+            nc_client_2 = await exit_stack.enter_async_context(
+                NetCatClient(
+                    connection,
                     serv_ip,
-                    str(80),
-                ]).run(
-                    stdout_callback=output_notifier.handle_output,
-                    stderr_callback=output_notifier.handle_output,
-                )
+                    80,
+                    ipv6=ip_proto == IPProto.IPv6,
+                    source_ip=alpha_ip,
+                ).run()
             )
 
-            await proc.wait_stdin_ready()
-            await asyncio.sleep(2.0)
-            # Without this sleep nc get EOF on stdin for some reason
-            await proc.write_stdin("GET")
-
             # Wait for both netcat processes
-            await sender_start_event.wait()
-            await sender_start_event.wait()
+            await asyncio.gather(
+                nc_client_1.connection_succeeded(), nc_client_2.connection_succeeded()
+            )
+
+            # exchange some data
+            await nc_client_2.send_data("GET")
+            await nc_client_2.receive_data()
```

Comment:
What is being received here by nc_client_2?

Reply:
Just waiting to receive the HTTP reply from the PHOTO_ALBUM server. It's not necessary for the test, I think it was just leftover checking if the method works, I can remove